packages/www/pages/dashboard/streams/[id]/index.tsx

```diff
@@ -38,39 +49,131 @@ const Overview = () => {
     refetchInterval,
   });
 
+  const doGetInfo = useCallback(
+    async (id: string) => {
+      setInfo(null);
+      const [, rinfo] = await getStreamInfo(id);
+      if (!rinfo || rinfo.isSession === undefined) {
+        return;
+      } else if (rinfo.stream) {
+        const info = rinfo as StreamInfo;
+        setInfo(info);
+      }
+    },
+    [getStreamInfo]
+  );
+
+  const getIngestRate = useCallback(
+    async (id: string) => {
+      const [, rinfo] = await getStreamInfo(id);
+      if (!rinfo?.session) {
+        return;
+      }
+      const newInfo = rinfo as StreamInfo;
+      setDataChart((prev) => {
+        const lastItem = prev[prev.length - 1];
+        return [
+          ...prev,
+          {
+            name: lastItem ? lastItem.name + ingestInterval / 1000 : 0,
+            kbps: Math.round((newInfo.session.ingestRate / 1000) * 8), // kilobits rather than bytes here
+          },
+        ].slice(Math.max(prev.length - maxItems, 0));
+      });
+    },
+    [getStreamInfo]
+  );
+
+  useEffect(() => {
+    if (!id) {
+      return;
+    }
+    doGetInfo(typeof id === "string" ? id : null);
+  }, [doGetInfo]);
+
+  useEffect(() => {
+    const intervalId = window.setInterval(() => {
+      if (info) {
+        getIngestRate(typeof id === "string" ? id : null);
```

Comment:
Edit: Just noticed that this was copied verbatim from the previous stream health implementation. I guess it's not necessary to change it but would still be a welcome addition/improvement 😄 

I don't think we should ever be calling neither `doGetInfo` nor `getIngestRate` with `null` as the input. That means they will do an useless roundtrip on the server just to get nothing back. If there's no use for that call, I think you should just check that `typeof` on the `if` that you have right before calling them. Like `if (typeof id !== "string") return`
```suggestion
    if (typeof id !== "string") {
      return;
    }
    doGetInfo(id);
  }, [doGetInfo]);

  useEffect(() => {
    const intervalId = window.setInterval(() => {
      if (typeof id === "string") {
        getIngestRate(id);
```
(I think the `if` before `getIngestRate` was wrong? Should it really be checking that `info` var instead which is not even used here?)